scripts/init_hdfs.sh

```diff
@@ -12,8 +21,15 @@ done
 
 # Set attributes on root directory.  It needs to be world writable so that
 #  other processes (e.g., solr, spark) can create directories for their storage.
-log INFO "Setting root directory permissions"
-bin/hdfs dfs -chmod 777 /
+log INFO "Set root directory permissions"
+hdfs dfs $HADOOP_OPTS -chmod 777 /
+
+log INFO "Set root directory ownership"
+hdfs dfs $HADOOP_OPTS -chown hdfs:hdfs /
+
+log INFO "Terminate namenode"
+kill %1
```

Comment:
Are you sure this is safe?  I only ask because apparently Solr doesn't shutdown cleanly in response to SIGTERM, so just thought I would double-check for HDFS.

Reply:
Fair question.  A bit of googling yields this command as the preferred way to shut down hdfs running as a daemon:
```
$HADOOP_PREFIX/sbin/hadoop-daemon.sh --config $HADOOP_CONF_DIR --script hdfs stop namenode
```
That command [issues a SIGTERM](https://github.com/apache/hadoop/blob/master/hadoop-common-project/hadoop-common/src/main/bin/hadoop-daemon.sh#L192), so I think it's safe to do so.